packages/esbuild/esbuild.bzl

```diff
@@ -273,14 +283,16 @@ For further information about esbuild, see https://esbuild.github.io/
     """,
 )
 
-def esbuild_macro(name, output_dir = False, **kwargs):
+def esbuild_macro(name, output_dir = False, output_css = False, **kwargs):
```

Comment:
Now after the rebase, what happens if the user passes `output = "foo.js"` when the rule is named `bar`? I'm guessing esbuild will output `foo.css` and bazel wont find the output?

I guess now we have the same issue with `output_map` // cc @mattsoulanille

Reply:
I don't think we can rely on the output file being split_extension()-able. I'm pretty sure the user is allowed to provide a file that has no extension at all, but I haven't tested it.

@mattem I think we need to remove the output_map arg. Esbuild (as far as I can tell) doesn't allow you to specify where the sourcemap gets created and always bases the path off of the output filename. The rule implementation also only uses output_map to tell bazel about the sourcemap and doesn't even pass it to esbuild. We should probably compute the path in the rule implementation instead of passing it as an arg.

Reply:
```
dae@dip:~% esbuild --bundle foo.js --outfile=bar.js

  bar.css  24b
  bar.js   15b

⚡ Done in 2ms
dae@dip:~% esbuild --bundle foo.js --outfile=bar

  bar.css  24b
  bar      15b

⚡ Done in 2ms
```

So it looks like stripping the extension could work. I have no preference either way, so please just let me know whether you'd prefer the current boolean behaviour be fixed, or it changed to accept a filename directly.

Reply:
Ah so looks like esbuild will always correctly identify the filename and append `.css`, so if there is an extension we can strip it and use it for the `output_css` name, rather than a boolean. It would be good if the two properties in the API had the same semantics.

@mattsoulanille True, however it also serves as a label that can be then addressed directly, eg a user could have `//foo:bar.js.map` as an input to a downstream rule. This comes in handy when you need to route both the `.js` and `.js.map` somewhere, but the `.css` to another rule rather than depending on the full `/foo:bundle` target.

What we should do though is remove the `.metadata` and `.jsconfig` from the default output 😅

Reply:
Sorry, I'm a bit confused - you talk about stripping, but also removing the boolean, whereas I would have thought stripping is only useful in the boolean case. Which would you prefer? :-)

Reply:
Oops sorry. Upon thinking about it a bit more, perhaps we don't do anything in the macro at all, and leave it completely up to the user to provide the correct name via `output_css` from the rule, and pass it with everything else via `**kwargs`. This way, if esbuild changes to allow control over the resulting css filename there's no change to our API, we would just have to pass an additional flag with the existing `output_css` name that we get, and the user can simply customize it by changing this existing attribute. wdyt? Sorry, I know we've gone around the houses here!

Reply:
No problem, done. If you'd like all the changes squashed into a single commit, please let me know.